crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -306,7 +306,28 @@ impl ShaderCache {
                             },
                         )?;
 
-                        ShaderSource::Naga(Cow::Owned(naga))
+                        #[cfg(feature = "coupled_naga")]
+                        {
+                            ShaderSource::Naga(Cow::Owned(naga))
+                        }
+
+                        #[cfg(not(feature = "coupled_naga"))]
+                        {
+                            let mut validator = naga::valid::Validator::new(
+                                naga::valid::ValidationFlags::empty(),
```

Comment:
The validation should likely be on, just to ensure the generated wgsl isn't bunk